### PR TITLE
 Add smart annotation to comments [SCI-3458]

### PR DESCRIPTION
**Pull request by aignatov-bio**
Jira ticket: [SCI-3458](https://biosistemika.atlassian.net/browse/SCI-3458)

### What was done
Create new view element for inline editing to show Smart Annotation objects. Update logic of inline editing.

### CODE REVIEW

File: app/assets/javascripts/shared/inline_editing.js [hunk 1]
```diff
@@ -63,21 +74,26 @@ function initInlineEditing(title) {
...
     }
 
-    $editBlock.click(e => {
+    $editBlock.click((e) => {
+      if (e.target.tagName === 'A') return true;
       if (inputString.disabled) {
         saveAllEditFields();
         editBlock.dataset.editMode = 1;
         inputString.disabled = false;
+        $inputString.removeClass('hidden').prev().addClass('hidden');
```
Comment (jbargu) [inline app/assets/javascripts/shared/inline_editing.js]: All this `prev` without selector is calling for some troubles down the road (not only here). Would it be possible to add selector inside so it's more readable for someone what this is doing, or maybe also add a comment

File: app/assets/javascripts/shared/inline_editing.js [hunk 2]
```diff
@@ -63,21 +74,26 @@ function initInlineEditing(title) {
       return true;
     }
 
-    $editBlock.click(e => {
+    $editBlock.click((e) => {
+      if (e.target.tagName === 'A') return true;
```
Comment (jbargu) [inline app/assets/javascripts/shared/inline_editing.js]: Please comment this, or it's hocus pocus. Which element is 'A'?

File: app/assets/javascripts/shared/inline_editing.js [hunk 3]
```diff
@@ -42,9 +45,17 @@ function initInlineEditing(title) {
         type: 'PUT',
         dataType: 'json',
         data: params,
-        success: function() {
+        success: function(result) {
+          var viewData;
+          if (editBlock.dataset.responseField) {
```
Comment (jbargu) [inline app/assets/javascripts/shared/inline_editing.js]: In what circumstance does this `if` make sense. Basically we just ignore the successful update?

File: app/assets/javascripts/shared/inline_editing.js (hunk 1 above)
Comment (aignatov-bio) [reply app/assets/javascripts/shared/inline_editing.js]: Move it to another line with selector

File: app/assets/javascripts/shared/inline_editing.js (hunk 2 above)
Comment (aignatov-bio) [reply app/assets/javascripts/shared/inline_editing.js]: 'A' it is tag name of <a></a> element

File: app/assets/javascripts/shared/inline_editing.js (hunk 3 above)
Comment (aignatov-bio) [reply app/assets/javascripts/shared/inline_editing.js]: By default we take string from input element, but if we want to modify this string on backend, we need to provide this data field.
In our case case in input string we must save raw text, but in preview show with html tags.

**Pull request merged**